src/core/VarReplay.jl

```diff
@@ -207,6 +207,9 @@ end
 Base.getindex(vi::VarInfo, vview::VarView) = copy(getval(vi, vview))
 Base.setindex!(vi::VarInfo, val::Any, vview::VarView) = setval!(vi, val, vview)
 
+Base.getindex(vi::VarInfo, s::Selector) = copy(getval(vi, getranges(vi, s)))
+Base.setindex!(vi::VarInfo, val::Any, s::Selector) = setval!(vi, val, getranges(vi, s))
```

Comment:
Looks identical to the next one, so use `Union{Selector, Sampler}` instead to avoid duplication.

Reply:
Fixed by #726.